tests/cupy_tests/manipulation_tests/test_join.py

```diff
@@ -51,6 +51,14 @@ def test_concatenate2(self, xp, dtype):
         c = testing.shaped_arange((2, 3, 3), xp, dtype)
         return xp.concatenate((a, b, c), axis=-1)
 
+    @testing.for_all_dtypes(name='dtype')
+    @testing.numpy_cupy_array_equal()
+    def test_concatenate_axis_none(self, xp, dtype):
+        a = testing.shaped_arange((2, 3, 4), xp, dtype)
+        b = testing.shaped_reverse_arange((2, 3, 2), xp, dtype)
+        c = testing.shaped_arange((2, 3, 3), xp, dtype)
+        return xp.concatenate((a, b, c), axis=None)
```

Comment:
All inputs must have the same `ndim` value when `axis` is NOT `None`, but don't need to do when `axis` is `None`. Could you add a test which checks `cupy.concatenate` allows different `ndim` and `shape`?